models/issue_comment.go

```diff
@@ -749,6 +749,9 @@ func createIssueDependencyComment(e *xorm.Session, doer *User, issue *Issue, dep
 	if !add {
 		cType = CommentTypeRemoveDependency
 	}
+	if err = issue.LoadRepo(); err != nil {
```

Comment:
You must load the repo from with the provided session `e`

Reply:
To be more explicit, when there is an open transaction you **MUST** use the provided `*xorm.Session` to do database lookups.

This line should read:

```go
     if err = issue.loadRepo(e); err != nil {
```

If you don't do this you will cause deadlocks.

Reply:
No problems. It was caught. That's the point of the review system. I've added bugs myself - see the debacle about the last migration. (Although I've more than suffered trying to create migration tests as my penance.)

Thanks for updating the tests too. I think they're due a rationalisation - we probably need to make the integration tests more story like - create repos, fix issues, upload stuff etc. At present they're more of a hodgepodge of almost unit tests which take too long to complete.